scarlet2/nn.py

```diff
@@ -0,0 +1,156 @@
+# -------------------------------------------------- #
+# This class is used to calculate the gradient       #
+# of the log-probability via calling the nn prior    #
+# ScoreNet model. A custom vjp is created to return  #
+# the nn prior when calling jax.grad()               #
+# -------------------------------------------------- #
+
+try:
+    import numpyro.distributions as dist
+    import numpyro.distributions.constraints as constraints
+    from galaxygrad import HSC_ScoreNet32, HSC_ScoreNet64, ZTF_ScoreNet32, ZTF_ScoreNet64,HSC_LogNet32, HSC_LogNet64# (https://pypi.org/project/galaxygrad/0.0.19)
```

Comment:
I don't think this should be a required import. This class needs to define the linkage between galaxygrad and scarlet2, but a user should import whatever model from galaxygrad they want to use.